lib/lcevc/lcevc_dil.js

```diff
@@ -0,0 +1,277 @@
+        this.dil_ = new LcevcDil.LcevcDil(
+            this.media_,
+            this.canvas_,
+            this.dilConfig_);
+      }
+    }
+  }
+
+  /**
+   * Close LCEVC Dil.
+   * @override
+   * @export
+   */
+  release() {
+    this.hideCanvas();
+    if (this.dil_) {
+      this.dil_.close();
+      this.dil_ = null;
+    }
+    if (this.lcevcContainer_) {
+      this.lcevcContainer_.remove();
+    }
+    if (this.resizeObserver_) {
+      this.resizeObserver_.disconnect();
+      this.resizeObserver_ = null;
+    }
+    if (this.canvas_) {
+      this.canvas_.remove();
+    }
+  }
+
+  /**
+   * Check if the LCEVC Dil lib is present and is supported by the browser.
+   * @return {boolean}
+   * @private
+   */
+  isSupported_() {
+    if (typeof libDPIModule === 'undefined') {
+      shaka.log.alwaysWarn(
+          'Could not Find LCEVC Library dependencies on this page');
+    }
+
+    if (typeof LcevcDil === 'undefined') {
+      shaka.log.alwaysWarn('Could not Find LCEVC Library on this page');
+    } else {
+      if (!LcevcDil.SupportObject.SupportStatus) {
+        shaka.log.alwaysWarn(LcevcDil.SupportObject.SupportError);
+      }
+    }
+
+    return typeof LcevcDil !== 'undefined' &&
+      typeof libDPIModule !== 'undefined' &&
+      LcevcDil.SupportObject.SupportStatus;
+  }
+
+  /**
+   * Create canvas for LCEVC Dil to render if not exists.
+   * @private
+   */
+  createCanvas_() {
+    this.canvas_ = /** @type {HTMLCanvasElement} */
+      (shaka.util.Dom.createHTMLElement('canvas'));
+    if (LcevcDil.SupportObject.webGLSupport(this.canvas_)) {
+      this.lcevcContainer_ = shaka.util.Dom.createHTMLElement('div');
+      this.lcevcContainer_.style.position = 'relative';
+
+      this.canvas_.style.position = 'absolute';
+      this.media_.style.display = 'none';
+
+      this.parent_ = this.media_.parentElement;
+      if (this.parent_) {
+        this.lcevcContainer_.appendChild(this.canvas_);
+        this.parent_.appendChild(this.lcevcContainer_);
+        if ('ResizeObserver' in window) {
+          this.resizeObserver_ = new ResizeObserver((_entries) => {
+            window.requestAnimationFrame(() => {
+              this.playerResize_();
+            });
+          });
+          this.resizeObserver_.observe(this.parent_);
+        }
+        this.playerResize_();
+      }
+    }
+  }
+
+  /**
+   * Resizing the container based on the source video dimensions.
+   * @private
+   */
+  playerResize_() {
+    const lcevcContainerParent = /** @type {HTMLElement} */
+      (this.lcevcContainer_.parentElement);
+
+    if (this.isSupported_()) {
+      // We try to find the actual aspect ratio with the lcevc enhanced frame
+      // if this value is not a valid value we default it to 1920x1080
+      const aspectRatio = (this.dil_ && this.dil_.aspectRatio > 0 ?
+        this.dil_.aspectRatio :
+        this.media_.videoWidth / this.media_.videoHeight) || 1920 / 1080;
+      this.lcevcContainer_.style.width =
```

Comment:
When you move this code to the UI layer, you don't need to set styles with an explicit width.  Instead, use a layout in CSS that overlays the video and matches 100% of the size of the video.  See styles in ui/less/, specifically the way shaka-text-container and shaka-controls-container overlay the video element and fill the same space.

In fact, if you do it this way, you really don't even need to hide the video element.  Making the canvas visible will simply cover up the video element.

Reply:
Created a new shaka-canvas-container class to be added to the canvas and append it to the videocontainer.